### PR TITLE
Update eslint to 3.8.1, noms was broken at 3.8.1

### PR DESCRIPTION
**Pull request by kalman**
Also remove roll.py (not used anymore, we use file path) and make it
pinned to a specific version (so that this doesn't happen again.

### CODE REVIEW

File: js/noms/src/json-convert.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import {
 import type Value from './value.js';
 
 type JSON = string | number | boolean | null | JSONObject | JSONArray;
-type JSONObject = { [key:string]: JSON };
+type JSONObject = { [key: string]: JSON };
```
Comment (arv) [inline js/noms/src/json-convert.js]: Ha! This still does not catch the lint error.

Comment (kalman) [inline js/noms/src/json-convert.js]: wat?

Comment (kalman) [inline js/noms/src/json-convert.js]: By "this" do you mean 3.7.1? Not sure what your comment means exactly :-) anyhow, submitting.

**Comment by arv** [comment]
I meant that there should be no space after {.

Let me see if I can find a rule for that.

**Pull request merged**